eZ/Publish/Core/Helper/PreviewLocationProvider.php

```diff
@@ -50,14 +52,37 @@ public function __construct(
      *
      * If the content doesn't have a location nor a location draft, null is returned.
      *
+     * @deprecated Since 7.5.4, rather use {@see loadMainLocationByContent}.
+     *
```

Comment:
Missing `@throws`:
```suggestion
     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
```

Reply:
Copying over doc from inline used functions.... 😒  But ok, if you want to maintain that it's fine with me :P

Reply:
> Copying over doc from inline used functions....  But ok, if you want to maintain that it's fine with me :P

The point here is that when not present, consumer won't know that such exception should be handled. Consumer should not look into the implementation to infer that :P

Reply:
True, but for generic use, using the `@uses` annotations solves that without having to maintain the doc for inline use twice _(because you will for sure forget to update this if the API doc changes ;) )_

Reply:
> True, but for generic use, using the `@uses` annotations solves that without having to maintain the doc for inline use twice _(because you will for sure forget to update this if the API doc changes ;) )_

Sorry to continue this not very important thread, but I want to make sure we're on the same page.
What I meant as a DX is to have something like this:
![image](https://user-images.githubusercontent.com/7099219/63274558-47ff9980-c2a0-11e9-90d0-c75165c456a2.png)

This won't show up until we document `@throws` on the actual method used in a custom code. Placing `@uses` in the called method doc does not make IDE "see" possible exceptions thrown by inner method calls.

I already see that I missed the fact that we should foremost do it on the `loadMainLocationByContent` not just on the `loadMainLocation`, but there's no point on spending time now to improve some core helper, I just wanted to show the issue in general.